app/kubemci/pkg/ingress/ingresssyncer.go

```diff
@@ -0,0 +1,152 @@
+					}
+				} else {
+					fmt.Println("Will not overwrite the differing Ingress resource without the --force flag.")
+					err = multierror.Append(fmt.Errorf("will not overwrite Ingress resource in cluster '%v' without --force", cluster))
+				}
+			} else {
+				fmt.Println("Ingress already exists and matches; moving on.")
+			}
+		}
+	}
+	return ensuredClusters, err
+}
+
+func (i *IngressSyncer) DeleteIngress(ing *v1beta1.Ingress, clients map[string]kubeclient.Interface) error {
+	var err error
+	for cluster, client := range clients {
+		fmt.Printf("Deleting Ingress from cluster: %v...\n", cluster)
+		glog.V(3).Infof("Using namespace %s for ingress %s", ing.Namespace, ing.Name)
+		deleteErr := client.ExtensionsV1beta1().Ingresses(ing.Namespace).Delete(ing.Name, &metav1.DeleteOptions{})
+		glog.V(2).Infof("Error in deleting ingress %s: %s", ing.Name, deleteErr)
+		if deleteErr != nil {
+			if errors.IsNotFound(err) {
+				fmt.Println("Ingress doesnt exist; moving on.")
+				continue
+			} else {
+				err = multierror.Append(err, fmt.Errorf("Error in deleting ingress from cluster %s: %s", cluster, deleteErr))
+			}
+		}
+	}
+	return err
+}
+
+func UnmarshallAndApplyDefaults(filename, namespace string, ing *v1beta1.Ingress) error {
+	if err := unmarshall(filename, ing); err != nil {
+		return err
+	}
+	if ing.Namespace == "" {
+		if namespace == "" {
+			ing.Namespace = defaultIngressNamespace
+		} else {
+			ing.Namespace = namespace
+		}
+	} else if namespace != "" && ing.Namespace != namespace {
+		return fmt.Errorf("the namespace from the provided ingress %q does not match the namespace %q. You must pass '--namespace=%v' to perform this operation.", ing.Namespace, namespace, ing.Namespace)
+	}
+	ingAnnotations := annotations.IngAnnotations(ing.Annotations)
+	class := ingAnnotations.IngressClass()
+	if class == "" {
+		addAnnotation(ing, annotations.IngressClassKey, annotations.GceMultiIngressClass)
+		glog.V(3).Infof("Adding class annotation to be of type %v\n", annotations.GceMultiIngressClass)
+	} else if class != annotations.GceMultiIngressClass {
+		return fmt.Errorf("ingress class is %v, must be %v", class, annotations.GceMultiIngressClass)
+	}
+
+	return nil
+}
+
+func ApplyStaticIP(staticIPName string, ing *v1beta1.Ingress) error {
+	ingAnnotations := annotations.IngAnnotations(ing.Annotations)
+
+	if ingAnnotations == nil || ingAnnotations.StaticIPName() == "" {
+		if staticIPName == "" {
+			return fmt.Errorf("ingress spec must provide a Global Static IP through annotation of %q (alternatively, use --static-ip flag)", annotations.StaticIPNameKey)
+		}
+		addAnnotation(ing, annotations.StaticIPNameKey, staticIPName)
+	} else if staticIPName != "" && ingAnnotations.StaticIPName() != staticIPName {
+		return fmt.Errorf("the StaticIPName from the provided ingress %q does not match --static-ip=%v. You must pass '--static-ip=%v' to perform this operation.", ingAnnotations.StaticIPName(), staticIPName, ingAnnotations.StaticIPName())
```

Comment:
optional since this is just moving existing code: replace "StaticIPName" in the error message by annotation key since that is what user knows about.